src/app/panels/text-panel/text-panel.component.ts

```diff
@@ -66,6 +70,15 @@ export class TextPanelComponent implements OnInit, OnDestroy {
     shareReplay(1),
   );
 
+  public msDesc$ = this.evtModelService.editionSource$.pipe(
+    map(s => {
+      const msDesc = s.querySelector<HTMLElement>('msDesc');
+      const msDescParser = createParser(MsDescParser, parse);
+
+      return msDescParser.parse(msDesc);
```

Comment:
This should be handled by a parser. Why is there parser logic inside a component?

Reply:
Parser logic should never be in a component. Move it to evt-model-service.